rust/variable/variable_concept.rs

```diff
@@ -28,73 +28,56 @@ use crate::{
     variable::variable::validate_variable_name,
 };
 
-#[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
-pub enum Visibility {
-    Visible,
-    Invisible,
-}
-
 #[derive(Debug, Clone, Eq, Hash, PartialEq)]
 pub enum ConceptVariable {
-    Anonymous(Visibility),
-    Name(String),
+    Anonymous,
+    Hidden,
+    Named(String),
 }
```

Comment:
Simplified

Reply:
We can make this change, but we have to make clear we're changing a fundamental model of TypeQL: We used to have 'anonymous' and 'named' variables only - now we're introducing a third fundamental type of variable: 'hidden'.

I tried to think of further implications - only came up with that we will have to decide how we want to treat them on the backend later. We can also split them on the backend and use numbers `$1` for example for hidden, and `$_` for anonymous in the future. Otherwise we may want to retract this change but we can come back here

Reply:
A big conceptual change like this also has to be outlined clearly in the PR description.

Reply:
This is not a big conceptual or structural change. The change is 1-to-1 from `Anonymous(Visible)` and `Anonymous(Invisible)` to `Anonymous` and `Hidden` respectively.

Reply:
it is a big conceptual change - we had 2 types of variables, now we have 3, rather than 2 where one has 2 variations (invisible/visible). This kind of thing affects how we think of query structures in our heads, so it is important.

Example - a conjunction in my head is always made of variables. Any hidden variable is anonymous, because anonymous variables can't be referred to twice. Thus any generated variable is anonymous, and our conjunctions are made of named or anonymous variables only. We can abstract away having another type of variable that is generated.

We could discuss that we should be treating generated variables as a different _kind_ of variable, has you've done here. This is in my mind OK.